sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashOuterJoin.scala

```diff
@@ -57,13 +57,26 @@ case class BroadcastHashOuterJoin(
   override def requiredChildDistribution: Seq[Distribution] =
     UnspecifiedDistribution :: UnspecifiedDistribution :: Nil
 
+  /**
+   * Use lazy so that we won't do broadcast when calling explain but still cache the broadcast value
+   * for the same query.
+   */
   @transient
-  private val broadcastFuture = future {
-    // Note that we use .execute().collect() because we don't want to convert data to Scala types
-    val input: Array[InternalRow] = buildPlan.execute().map(_.copy()).collect()
-    val hashed = HashedRelation(input.iterator, buildKeyGenerator, input.size)
-    sparkContext.broadcast(hashed)
-  }(BroadcastHashOuterJoin.broadcastHashOuterJoinExecutionContext)
+  private lazy val broadcastFuture = {

```

Comment:
Although @mateiz removed `lazy` in #1990, I want to add it back here. Because without `lazy`, if the user runs `explain`, the `broadcastFuture` will run and I cannot track the job.